app/app_iot_core.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package app
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/pkg/errors"
+
+	"github.com/mendersoftware/iot-manager/client/iotcore"
+	"github.com/mendersoftware/iot-manager/model"
+)
+
+func (a *app) provisionIoTCoreDevice(
+	ctx context.Context,
+	deviceID string,
+	integration model.Integration,
+	device *iotcore.Device,
+) error {
+	accessKeyID := integration.Credentials.AccessKeyID
+	secretAccessKey := integration.Credentials.SecretAccessKey
+	if accessKeyID == nil || secretAccessKey == nil {
+		return ErrNoConnectionString
+	}
+
+	sess, _ := session.NewSession(&aws.Config{
+		Credentials: credentials.NewStaticCredentials(*accessKeyID, *secretAccessKey, ""),
+	})
+
+	dev, err := a.iotcoreClient.UpsertDevice(ctx, sess, deviceID, device)
+	if err != nil {
+		return errors.Wrap(err, "failed to update iotcore devices")
+	}
+
+	if dev.Certificate != "" && dev.PrivateKey != "" {
+		err = a.wf.ProvisionExternalDevice(ctx, dev.ID, map[string]string{
+			confKeyAWSCertificate: dev.Certificate,
+			confKeyAWSPrivateKey:  dev.PrivateKey,
+		})
```

Comment:
the private key will not land in the workflows logs, right? (just so I can sleep peacefully)

Reply:
Private key will land to workflows, and the get propagated to deviceconfig.
This is what we have for Azure IoT Hub as well (connection string).
We have an epic to manage secrets, but the current implementation requires this.